gap/PolygonalComplexes/animating.gd

```diff
@@ -0,0 +1,537 @@
+#! @Returns the updated print record.
+#! @Arguments surface, i, printRecord
+DeclareOperation( "ActivateInnerCircle", [IsSimplicialSurface and IsNotEdgeRamified, IsCyclotomic, IsRecord] );
+#! @Arguments surface, i, printRecord
+DeclareOperation( "DeactivateInnerCircle", [IsSimplicialSurface and IsNotEdgeRamified, IsCyclotomic, IsRecord] );
+#! @Arguments surface, i, printRecord
+DeclareOperation( "IsInnerCircleActive", [IsSimplicialSurface and IsNotEdgeRamified, IsCyclotomic, IsRecord] );
+#! @EndGroup
+
+#! @BeginGroup SetCircleColours
+#! @Description
+#! For the describtion of inner circles look at <Ref Sect="Section_LabelInnerCirclesAnimating"/>. Set colours for all inner circles. (Colours are stored in the format 0xABCDEF where A,B,C,D,E,F are elements of the hexadecimal code.)
+#! @Returns the updated print record.
+#! @Arguments surface, newColoursList, printRecord
+DeclareOperation( "SetCircleColours", [IsSimplicialSurface and IsNotEdgeRamified, IsDenseList, IsRecord] );
+#! @Arguments surface, printRecord
+DeclareOperation( "GetCircleColours", [IsSimplicialSurface and IsNotEdgeRamified, IsRecord] );
+#! @EndGroup
+
+#! @BeginGroup SetCircleColour
+#! @Description
+#! For the describtion of inner circles look at <Ref Sect="Section_LabelInnerCirclesAnimating"/>. Set the <A>colour</A> of the inner circle corresponding to the face in position <A>i</A> of <A>Faces(surface)</A>. (Colours are stored in the format 0xABCDEF where A,B,C,D,E,F are elements of the hexadecimal code.)
+#! @Returns the updated print record.
+#! @Arguments surface, i, colour, printRecord
+DeclareOperation( "SetCircleColour", [IsSimplicialSurface and IsNotEdgeRamified, IsCyclotomic, IsString, IsRecord] );
+#! @Arguments surface, i, printRecord
+DeclareOperation( "GetCircleColour", [IsSimplicialSurface and IsNotEdgeRamified, IsCyclotomic, IsRecord] );
+#! @EndGroup
+
+#! @BeginGroup CalculateParametersOfInnerCircle
+#! @Description
+#! For the describtion of inner circles look at <Ref Sect="Section_LabelInnerCirclesAnimating"/>. This function calculates the parameters of the innerer circles (incenter, radius, angles relative to x-y-plane) and their normal vectors (angles relative to x-direction) based on the
+#! 3D-coordinates of the vertices and saves those to the print record.
+#! @Returns the updated print record.
+#! @Arguments surface, printRecord
+DeclareOperation( "CalculateParametersOfInnerCircle", [IsSimplicialSurface and IsNotEdgeRamified, IsRecord] );
+#! @EndGroup
+
+#! @Section Normales of Inner Circles
+#! @SectionLabel LabelNormalesInnerCirclesAnimating
+#!
+
+#! A normal of an inner circle is a vector (represented as a line) which goes orthogonal through the middle point of the corresponding inner circle. (For the describtion of inner circles look at <Ref Sect="Section_LabelInnerCirclesAnimating"/>.)
```

Comment:
... of an inner circle (compare REF) is a line intersecting the center of the circle orthogonally.